tls/s2n_server_finished.c

```diff
@@ -45,14 +45,10 @@ S2N_RESULT s2n_finished_recv(struct s2n_connection *conn, uint8_t *local_verify_
     return S2N_RESULT_OK;
 }
 
-S2N_RESULT s2n_finished_send(struct s2n_connection *conn, uint8_t *seq_num, uint8_t *verify_data)
+S2N_RESULT s2n_finished_send(struct s2n_connection *conn, uint8_t *verify_data)
 {
     RESULT_ENSURE_REF(conn);
 
-    struct s2n_blob seq = { 0 };
-    RESULT_GUARD_POSIX(s2n_blob_init(&seq, seq_num, S2N_TLS_SEQUENCE_NUM_LEN));
-    RESULT_GUARD_POSIX(s2n_blob_zero(&seq));
-
```

Comment:
Since this isn't related to the rest of this PR, and this PR is quite large, maybe move this to a separate PR? It is a small fix, but it should be able to stand on its own.